content/src/storage/ContentStorage.ts

```diff
@@ -1,24 +1,34 @@
-import { ContentFileHash } from 'dcl-catalyst-commons'
 import { Readable } from 'stream'
+import { createGunzip } from 'zlib'
 
 export type ContentEncoding = 'gzip'
 
 export interface ContentStorage {
-  storeStream(id: string, content: Readable): Promise<void>
-  /** @deprecated use storeStream instead */
-  store(id: string, content: Uint8Array): Promise<void>
-  delete(ids: string[]): Promise<void>
-  retrieve(id: string): Promise<ContentItem | undefined>
-  exist(ids: string[]): Promise<Map<string, boolean>>
-  stats(id: string): Promise<{ size: number } | undefined>
-  storeContent(fileHash: ContentFileHash, content: Uint8Array | Readable): Promise<void>
-  size(fileHash: ContentFileHash): Promise<number | undefined>
+  storeStream(fileId: string, content: Readable): Promise<void>
```

Comment:
What about adding an `encoding` arg to signal if the content is already encoded or needs to be encoded before being stored, so we can get rid of the hack?
If not, the interface exposes the possibility to return a `RawStream` which could be encoded, but it doesn't have the ability to say that something is or should be encoded.

Reply:
adding the encoding may generate problems with the file names. I've added a `storeStreamAndCompress` to remove the hack and also comply with this review